controller/api/destination/server.go

```diff
@@ -277,11 +277,22 @@ func (s *server) getProfileByIP(
 
 	if svcID == nil {
 		// If the IP does not map to a service, check if it maps to a pod
-		pod, err := getPodByIP(s.k8sAPI, ip.String(), port, s.log)
+		var pod *corev1.Pod
+		targetIP := ip.String()
+		pod, err = getPodByPodIP(s.k8sAPI, ip.String(), port, s.log)
```

Comment:
Since we already do the string conversion, can we just use targetIP instead? Same for the call to `getPodByHostIP`.

```suggestion
		pod, err = getPodByPodIP(s.k8sAPI, targetIP, port, s.log)
```